### PR TITLE
SetComponentActivationAndSetOperatingMode and SetAlertActivationAndSe…

### PR DESCRIPTION
**Pull request by NerijusKondrotas**
…tOperatingMode manipulations

// Describe the purpose of the pull request here and remove this line. Keep the checklist.

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [x] Reviewer

### CODE REVIEW

File: src/t2iapi/combined/service.proto [hunk 1]
```diff
@@ -29,4 +29,26 @@ service CombinedService {
...
+  /*
+  Request to create a new or adapt an existing alert activation state to the given AlertActivation value and
+  set the requested @OperatingMode for the referenced operation.
+  The alert state @Handle which was created or adapted shall be returned.
+  The manipulated states shall be persistent until a next manipulation call or an SDC operation invocation.
+  If the device is not able to maintain the static states, it shall return RESULT_NOT_SUPPORTED.
+  */
+  rpc SetAlertActivationAndSetOperatingMode (
+  t2iapi.combined.SetAlertActivationAndSetOperatingModeRequest)
+      returns (t2iapi.context.SetAlertActivationAndSetOperatingModeResponse);
```
Comment (a-kleinf) [inline src/t2iapi/combined/service.proto:53]: must be BasicResponse

File: src/t2iapi/combined/combined_requests.proto [hunk 2]
```diff
@@ -29,3 +29,29 @@ message CreateContextStateWithAssociationAndSetOperatingModeRequest {
...
+}
+
+/*
+Request to create a new or adapt an existing alert activation state to the given AlertActivation value and
+set the requested @OperatingMode for the referenced operation.
+ */
+message SetAlertActivationAndSetOperatingModeRequest {
+    string alert_descriptor_handle = 1; // descriptor handle of the descriptor for which the ActivationState shall be
+                                        // set in the corresponding state element
+    AlertActivation activation = 2; // value to set as the ActivationState value
```
Comment (a-kleinf) [inline src/t2iapi/combined/combined_requests.proto:53]: must be t2iapi.activation_state.AlertActivation

File: src/t2iapi/combined/combined_requests.proto [hunk 3]
```diff
@@ -29,3 +29,29 @@ message CreateContextStateWithAssociationAndSetOperatingModeRequest {
...
 }
+
+/*
+Request to create a new or adapt an existing component activation state to the given ComponentActivation value and
+set the requested @OperatingMode for the referenced operation.
+ */
+message SetComponentActivationAndSetOperatingModeRequest {
+    string component_metric_descriptor_handle = 1; // descriptor handle of the descriptor for which the ActivationState
+                                                   // shall be set in the corresponding state element
+    ComponentActivation activation = 2; // value to set as the ActivationState value
```
Comment (a-kleinf) [inline src/t2iapi/combined/combined_requests.proto:40]: t2iapi.activation_state.ComponentActivation
and add import:
import "t2iapi/activation_state/types.proto";

File: src/t2iapi/combined/service.proto [hunk 4]
```diff
@@ -29,4 +29,26 @@ service CombinedService {
...
+  /*
+  Request to create a new or adapt an existing component activation state to the given ComponentActivation value and
+  set the requested @OperatingMode for the referenced operation.
+  The component state @Handle which was created or adapted shall be returned.
+  The manipulated states shall be persistent until a next manipulation call or an SDC operation invocation.
+  If the device is not able to maintain the static states, it shall return RESULT_NOT_SUPPORTED.
+  */
+  rpc SetComponentActivationAndSetOperatingMode (
+  t2iapi.combined.SetComponentActivationAndSetOperatingModeRequest)
+      returns (t2iapi.context.SetComponentActivationAndSetOperatingModeResponse);
```
[the remark this answers is not in the document]
Comment (a-kleinf) [reply src/t2iapi/combined/service.proto:42]: and import:
import "t2iapi/basic_responses.proto";

File: src/t2iapi/combined/combined_requests.proto [hunk 5]
```diff
@@ -29,3 +29,29 @@ message CreateContextStateWithAssociationAndSetOperatingModeRequest {
...
+    string operation_descriptor_handle = 3; // the operation descriptor handle for which the @OperatingMode value
+    // shall be set
+    t2iapi.operation.OperatingMode operating_mode = 4; // the requested @OperatingMode value of the operation state
+}
+
+/*
+Request to create a new or adapt an existing alert activation state to the given AlertActivation value and
+set the requested @OperatingMode for the referenced operation.
+ */
```
Comment (a-kleinf) [inline src/t2iapi/combined/combined_requests.proto:49]: adapt to something like:
/*
Request to set the ActivationState of the alert with the given handle to the requested AlertActivation value and to set the OperatingMode of the operation with the given handle to the requested OperatingMode value.
 */

File: src/t2iapi/combined/combined_requests.proto [hunk 6]
```diff
@@ -29,3 +29,29 @@ message CreateContextStateWithAssociationAndSetOperatingModeRequest {
                                             // shall be set
     t2iapi.operation.OperatingMode operating_mode = 4;  // the requested @OperatingMode value of the operation state
 }
+
+/*
+Request to create a new or adapt an existing component activation state to the given ComponentActivation value and
+set the requested @OperatingMode for the referenced operation.
+ */
```
Comment (a-kleinf) [inline src/t2iapi/combined/combined_requests.proto:36]: adapt to something like:
/*
Request to set the ActivationState of the component or metric with the given handle to the requested ComponentActivation value and to set the OperatingMode of the operation with the given handle to the requested OperatingMode value.
*/

File: src/t2iapi/combined/service.proto [hunk 7]
```diff
@@ -29,4 +29,26 @@ service CombinedService {
...
   t2iapi.combined.CreateContextStateWithAssociationAndSetOperatingModeRequest)
       returns (t2iapi.context.CreateContextStateWithAssociationResponse);
+
+  /*
+  Request to create a new or adapt an existing component activation state to the given ComponentActivation value and
+  set the requested @OperatingMode for the referenced operation.
+  The component state @Handle which was created or adapted shall be returned.
+  The manipulated states shall be persistent until a next manipulation call or an SDC operation invocation.
+  If the device is not able to maintain the static states, it shall return RESULT_NOT_SUPPORTED.
+  */
```
Comment (a-kleinf) [inline src/t2iapi/combined/service.proto:39]: use the description from the request and add
 /* The manipulated states shall be persistent until a next manipulation call or an SDC operation invocation.
  If the device is not able to maintain the static states, it shall return RESULT_NOT_SUPPORTED.*/

File: src/t2iapi/combined/service.proto [hunk 8]
```diff
@@ -29,4 +29,26 @@ service CombinedService {
...
+  t2iapi.combined.SetComponentActivationAndSetOperatingModeRequest)
+      returns (t2iapi.context.SetComponentActivationAndSetOperatingModeResponse);
+
+  /*
+  Request to create a new or adapt an existing alert activation state to the given AlertActivation value and
+  set the requested @OperatingMode for the referenced operation.
+  The alert state @Handle which was created or adapted shall be returned.
+  The manipulated states shall be persistent until a next manipulation call or an SDC operation invocation.
+  If the device is not able to maintain the static states, it shall return RESULT_NOT_SUPPORTED.
+  */
```
Comment (a-kleinf) [inline src/t2iapi/combined/service.proto:50]: use the description from the request and add
 /* The manipulated states shall be persistent until a next manipulation call or an SDC operation invocation.
  If the device is not able to maintain the static states, it shall return RESULT_NOT_SUPPORTED.*/

File: src/t2iapi/combined/combined_requests.proto [hunk 9]
```diff
@@ -29,3 +31,30 @@ message CreateContextStateWithAssociationAndSetOperatingModeRequest {
...
+to set the OperatingMode of the operation with the given handle to the requested OperatingMode value.
+*/
+message SetAlertActivationAndSetOperatingModeRequest {
+    string alert_descriptor_handle = 1; // descriptor handle of the descriptor for which the ActivationState shall be
+                                        // set in the corresponding state element
+    t2iapi.activation_state.AlertActivation activation= 2; // value to set as the ActivationState value
+    string operation_descriptor_handle = 3; // the operation descriptor handle for which the @OperatingMode value
+                                            // shall be set
+    t2iapi.operation.OperatingMode operating_mode = 4; // the requested @OperatingMode value of the operation state
+}
```
Comment (a-kleinf) [inline src/t2iapi/combined/combined_requests.proto:60]: in both new mesages - adapt the wordning of the attributes to be consistent, see also CreateContextStateWithAssociationAndSetOperatingModeRequest

**Review by a-kleinf — CHANGES_REQUESTED** [review]

**Review by a-kleinf — APPROVED** [review]

**Review by ldeichmann — APPROVED** [review]

**Pull request merged**